packages/manager/apps/hpc-vmware-public-vcf-aas/src/pages/dashboard/datacentre/edge-gateway/edit/EditEdgeGateway.page.tsx

```diff
@@ -0,0 +1,111 @@
+import { Suspense, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useNavigate, useParams } from 'react-router-dom';
+import { Controller, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { NAMESPACES } from '@ovh-ux/manager-common-translations';
+import {
+  GetEdgeGatewayParams,
+  useUpdateEdgeGateway,
+  useVcdEdgeGatewayMocks,
+} from '@ovh-ux/manager-module-vcd-api';
+import { Drawer } from '@ovh-ux/manager-react-components';
+import { useMessageContext } from '@/context/Message.context';
+import { AddEdgeForm, EDGE_FORM_SCHEMA } from '@/schemas/edge.schema';
+import { InputField } from '@/components/form/InputField.component';
+import { SelectField } from '@/components/form/SelectField.component';
+import { EDGE_GATEWAY_NAME_MAX_LENGTH } from '../add/adgeEdgeGateway.constants';
+import Loading from '@/components/loading/Loading.component';
+
+export default function EditEdgeGatewayPage() {
+  const { t } = useTranslation('datacentres/edge-gateway');
+  const { t: tActions } = useTranslation(NAMESPACES.ACTIONS);
+  const { id, vdcId, edgeGatewayId } = useParams();
+  const navigate = useNavigate();
+  const closeDrawer = () => navigate('..');
+  const { addSuccess, addError } = useMessageContext();
+
+  const edgeParams: GetEdgeGatewayParams = { id, vdcId, edgeGatewayId };
+  const { data: edge, isLoading } = useVcdEdgeGatewayMocks(edgeParams);
+
+  const {
+    mutate: updateEdgeGateway,
+    isPending: isUpdating,
+  } = useUpdateEdgeGateway({
+    ...edgeParams,
+    onSettled: closeDrawer,
+    onSuccess: () => addSuccess({ content: t('edge_update_banner_success') }),
+    onError: () => addError({ content: t('edge_operation_error') }),
+  });
+
+  const {
+    control,
+    handleSubmit,
+    formState: { isValid },
+    reset,
+  } = useForm<AddEdgeForm>({
+    mode: 'onTouched',
+    resolver: zodResolver(EDGE_FORM_SCHEMA),
+    defaultValues: { edgeGatewayName: '', ipBlock: '' },
+  });
+
+  const onSubmit = (data: AddEdgeForm) =>
+    updateEdgeGateway({ edgeGatewayNewName: data.edgeGatewayName });
+
+  useEffect(() => {
+    if (!isLoading && edge?.currentState) {
+      reset({
+        edgeGatewayName: edge.currentState.edgeGatewayName,
+        ipBlock: edge.currentState.ipBlock,
+      });
+    }
+  }, [isLoading, edge]);
```

Comment:
This is not good and it is the fault of the guy who designed the drawer: me.
This useEffect could be avoided by putting the form in a child component and loading the data in the parent, but the Drawer does not allow this for now.
A new version of the Drawer will come with MUK, with composable parts, so you will be able to create a child component with a "Drawer.Content" and a "Drawer.Footer" imported from MUK.

[I did something like this here](https://github.com/ovh/manager/blob/dev/okms-01/packages/manager/apps/okms/src/modules/secret-manager/pages/drawers/create-version-drawer/CreateVersionDrawer.page.tsx). I add to recreate the inner parts of the drawer in the uApp, and they will be replaced by MUK components when we will do the migration.